nomad/job_endpoint_hooks.go

```diff
@@ -415,6 +421,17 @@ func (v *jobValidate) validateVaultIdentity(t *structs.Task) ([]error, error) {
 		return warnings, nil
 	}
 
+	// Only the default cluster is allowed to use the legacy flow. Using a
+	// non-default cluster requires an identity to be provided either in the
+	// task or as a default in the agent config.
+	if t.Vault.Cluster != structs.VaultDefaultCluster {
+		mErr = multierror.Append(mErr, fmt.Errorf(
```

Comment:
This is making me realize we're not taking advantage of the multierror in this method at all. We always return for the first error.

Reply:
Hum....true. This logic used to be _a lot_ more complex before, so I think it may have cases where multiple errors were returned. But now I think we can return a single error.

Another thing I'm realizing is that we run mutators before validators, so the `implicitiy_identities` hook should have already handled the agent config part, so I think we can only validate the task here.

Reply:
Fixed in https://github.com/hashicorp/nomad/pull/18810/commits/ace07aa14843223106167becba1f8d3462cc4b0a